runtime/common/src/main/java/edu/snu/nemo/runtime/common/plan/physical/PhysicalPlanGenerator.java

```diff
@@ -62,6 +64,22 @@ private PhysicalPlanGenerator(@Parameter(JobConf.DAGDirectory.class) final Strin
   public DAG<PhysicalStage, PhysicalStageEdge> apply(final DAG<IRVertex, IREdge> irDAG) {
     // first, stage-partition the IR DAG.
     final DAG<Stage, StageEdge> dagOfStages = stagePartitionIrDAG(irDAG);
+
+    final Map<String, List<StageEdge>> invariantIrEdgeMap = new HashMap<>();
+
+    dagOfStages.topologicalDo(irVertex -> dagOfStages.getIncomingEdgesOf(irVertex).stream().forEach(e -> {
+      final Pair<Integer, String> invariantProperty = e.getProperty(ExecutionProperty.Key.InvariantData);
+      if (invariantProperty != null) {
+        invariantIrEdgeMap.computeIfAbsent(invariantProperty.right(), k -> new ArrayList<>()).add(e);
+      }
+    }));
+
+    invariantIrEdgeMap.forEach((id, edges) -> {
+      final StageEdge baseEdge = edges.get(0);
+      final Pair<Integer, String> baseProperty = baseEdge.getProperty(ExecutionProperty.Key.InvariantData);
+      edges.forEach(e ->
+          e.getExecutionProperties().put(InvariantDataProperty.of(Pair.of(baseProperty.left(), baseEdge.getId()))));
```

Comment:
I suppose the LOOP_DUPLICATE_DATA_ID becomes an id of a physical edge here.
My understanding is that the `InvariantDataProperty` goes through three phases

1) (1, LOOP_DUPLICATE_DATA_ID)
2) (number of duplicate edges, LOOP_DUPLICATE_DATA_ID)
3) (number_of_duplicate_edges, physical edge id)

I'd think providing a separate property for each phase would make each phase more easy to understand.